WordPressAuthenticator/Analytics/SignInTracker.swift

```diff
@@ -0,0 +1,341 @@
+import Foundation
+
+/// Implements the analytics tracking logic for our sign in flow.
+///
+public class SignInTracker {
+    
+    /// The method used for analytics tracking.  Useful for overriding in automated tests.
+    ///
+    typealias TrackerMethod = (_ event: AnalyticsEvent) -> ()
+
+    enum EventType: String {
+        case step = "unified_login_step"
+        case interaction = "unified_login_interaction"
+        case failure = "unified_login_failure"
+    }
+    
+    enum Property: String {
+        case failure
+        case flow
+        case click
+        case source
+        case step
+    }
+    
+    enum Source: String {
+        case `default`
+        case jetpack
+        case share
+        case deeplink
+        case reauthetication
+        
+        /// Starts when the used adds a site from the site picker
+        ///
+        case selfHosted
+    }
+    
+    enum Flow: String {
+        /// The initial flow before we decide whether the user is logging in or signing up
+        case wpCom = "wordpress_com"
+        
+        /// The flow that starts when the user starts the Google login
+        ///
+        case googleLogin = "google_login"
+        
+        case googleSignup = "google_signup"
```

Comment:
I don't see `google_signup` mentioned in pbArwn-AP-p2. I think this one needs deleted in favor of `signup` ?

Reply:
This came up in a chat with @renanferrari .  Ref: pbArwn-AP-p2#comment-1160

Reply:
Ugh I never hit send on that comment. Ok, this is going to add a few small steps to this PR.

Reply:
Just want to double check I got this right.  I've added documentation to this, and the event is coming from this comment: pbArwn-AP-p2#comment-1160

Does that resolve?

Reply:
Yep! That is correct.